MdeModulePkg/Application/UiApp/FrontPage.c

```diff
@@ -421,24 +421,31 @@ ConvertProcessorToString (
 
 
 /**
-  Convert Memory Size to a string.
+  Convert Memory Info to a string.
 
   @param MemorySize      The size of the memory to process
+  @param MemorySpeed     The speed of the memory to process
   @param String          The string that is created
 
 **/
 VOID
-ConvertMemorySizeToString (
+ConvertMemoryInfoToString (
   IN  UINT32          MemorySize,
+  IN  UINT16          MemorySpeed,
   OUT CHAR16          **String
   )
 {
   CHAR16  *StringBuffer;
+  CHAR16  SpeedBuffer[16];
 
-  StringBuffer = AllocateZeroPool (0x24);
+  UnicodeValueToStringS (SpeedBuffer, 16, LEFT_JUSTIFY, MemorySpeed, 10);
+
+  StringBuffer = AllocateZeroPool (0x44);
   ASSERT (StringBuffer != NULL);
-  UnicodeValueToStringS (StringBuffer, 0x24, LEFT_JUSTIFY, MemorySize, 10);
-  StrCatS (StringBuffer, 0x24 / sizeof (CHAR16), L" MB RAM");
+  UnicodeValueToStringS (StringBuffer, 0x44, LEFT_JUSTIFY, MemorySize, 10);
+  StrCatS (StringBuffer, 0x44 / sizeof (CHAR16), L" MB RAM @ ");
+  StrCatS (StringBuffer, 0x44 / sizeof (CHAR16), SpeedBuffer);
+  StrCatS (StringBuffer, 0x44 / sizeof (CHAR16), L" MHz");

```

Comment:
Can we omit displaying the memory speed if it is 0? I recall not all FSPs report such information in the memory HOB properly.

Reply:
Sure: https://github.com/Dasharo/edk2/compare/2b296cd9b3b8aba29fb74cb3aa26fa4dd1003c1b..8d5c529dacf0690c419f4620755f13620071295b#diff-4cd8af86833dfa700ee981c69dc8c4c187dbd6451fd5ab372d7dd70602f51b73